flutter/lib/default_integrations.dart

```diff
@@ -84,6 +84,49 @@ Integration runZonedGuardedIntegration(
   return integration;
 }
 
+Integration deviceInfosIntegration(
+  SentryOptions options,
+  MethodChannel channel,
+) {
+  Future<void> integration(Hub hub, SentryOptions options) async {
+    try {
+      final Map<String, dynamic> infos =
+          Map<String, dynamic>.from(await channel.invokeMethod('deviceInfos'));
+
+      if (infos['contexts'] != null) {
+        options.addEventProcessor((event, dynamic hint) {
+          event = event.copyWith(
+            contexts: Contexts.fromJson(
+              Map<String, dynamic>.from(infos['contexts'] as Map),
+            ),
+          );
```

Comment:
this would be executed for Android too, so the `Android` and `Web` plugins need to return empty instead of non implemented method

Reply:
also, the result should be `merged` and not `overwritten`, if the same key already exists in the event, the event has precedence over the gotten contexts

Reply:
✅